app/src/main/java/comcmput301f16t01/github/carrier/MakeRequestActivity.java

```diff
@@ -6,33 +6,119 @@
 import android.support.v7.app.AlertDialog;
 import android.support.v7.app.AppCompatActivity;
 import android.os.Bundle;
+import android.view.MotionEvent;
 import android.view.View;
 import android.view.WindowManager;
-import android.widget.Button;
 import android.widget.EditText;
+import android.widget.ImageButton;
 import android.widget.TextView;
 import android.widget.Toast;
-
-import org.w3c.dom.Text;
-
 import java.util.Locale;
+import android.os.Handler;
+
+/*
+ The code for incrementing/decrementing the fare while holding down
+ the up and down arrows is based on: https://goo.gl/zKpYnX
+ Author: Yar
+ Retrieved on: November 5th, 2016
+  */
 
 public class MakeRequestActivity extends AppCompatActivity {
 
     final Activity activity = MakeRequestActivity.this;
+    final int REP_DEL = 50;
+
     private Location start = null;
     private Location end = null;
     private int fareEstimated = -1;
 
+    private Handler repeatUpdateHandler = new Handler();
+    private boolean autoIncrement = false;
+    private boolean autoDecrement = false;
+
+    class RepeatUpdater implements Runnable {
+        public void run() {
+            if(autoIncrement) {
+                incrementFare();
+                repeatUpdateHandler.postDelayed( new RepeatUpdater(), REP_DEL);
+            } else if(autoDecrement) {
+                decrementFare();
+                repeatUpdateHandler.postDelayed( new RepeatUpdater(), REP_DEL);
+            }
+        }
+    }
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_make_request);
         setTitle("New Request");
 
+        setButtons();
+
         activity.getWindow().setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_ADJUST_PAN);
     }
 
+    private void setButtons() {
+        ImageButton fareUpButton = (ImageButton) findViewById(R.id.imageButton_fareUp);
+        fareUpButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                if(fareEstimated != -1) {
+                    incrementFare();
+                }
+            }
```

Comment:
I did have like that initially. I made it more complicated because in a previous version I needed to call incrementFare with no parameters in one situation and with a View parameter from android:onClick so it was easier to have one function. I changed other things so this is possible now. I will make the change.